SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+//
+//  SDLSecondaryTransportManager.m
+//  SmartDeviceLink
+//
+//  Created by Sho Amano on 2018/02/28.
+//  Copyright © 2018 Xevo Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "SDLSecondaryTransportManager.h"
+
+#import "SDLControlFramePayloadConstants.h"
+#import "SDLControlFramePayloadRPCStartServiceAck.h"
+#import "SDLControlFramePayloadTransportEventUpdate.h"
+#import "SDLIAPTransport.h"
+#import "SDLLogMacros.h"
+#import "SDLNotificationConstants.h"
+#import "SDLProtocol.h"
+#import "SDLProtocolHeader.h"
+#import "SDLProtocolMessage.h"
+#import "SDLSecondaryTransportManagerConstants.h"
+#import "SDLStateMachine.h"
+#import "SDLTCPTransport.h"
+#import "SDLTimer.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+// same as SDLProtocol.m
+typedef NSNumber SDLServiceTypeBox;
+
+typedef NS_ENUM(NSInteger, SDLTransportClass) {
+    SDLTransportClassInvalid = 0,
+    SDLTransportClassPrimary = 1,
+    SDLTransportClassSecondary = 2,
+};
+typedef NSNumber SDLTransportClassBox;
+
+typedef NS_ENUM(NSInteger, SDLTransportSelection) {
+    SDLTransportSelectionDisabled,   // only for Secondary Transport
+    SDLTransportSelectionIAP,
+    SDLTransportSelectionTCP
+};
+typedef NSNumber SDLTransportSelectionBox;
+
+SDLSecondaryTransportState *const SDLSecondaryTransportStateStopped = @"Stopped";
+SDLSecondaryTransportState *const SDLSecondaryTransportStateStarted = @"Started";
+SDLSecondaryTransportState *const SDLSecondaryTransportStateConfigured = @"Configured";
+SDLSecondaryTransportState *const SDLSecondaryTransportStateConnecting = @"Connecting";
+SDLSecondaryTransportState *const SDLSecondaryTransportStateRegistered = @"Registered";
+SDLSecondaryTransportState *const SDLSecondaryTransportStateReconnecting = @"Reconnecting";
+
+/// Timeout for receiving Register Secondary Transport ACK frame
+static const float RegisterTransportTime = 10.0;
+
+/// Delay to retry connection after secondary transport is disconnected or registration is failed
+static const float RetryConnectionDelay = 15.0;
+
+
+/// An internal class to receive event from primary transport.
+@interface PrimaryProtocolListener : NSObject <SDLProtocolListener>
```

Comment:
Every class should have its own file (outside of perhaps extraordinarily small, struct-like ones on a case-by-case basis).

Reply:
I noticed your another comment https://github.com/smartdevicelink/sdl_ios/pull/947#discussion_r205576128 after I did this change. Let me think about it again.

Reply:
The class is once again split into dedicated file in commit f21418021582633997b7684b0ce128c88dfe3cac.